### PR TITLE
fix: libuv can return UV_FILE or UV_TTY for FILE_TYPE_CHAR handles

### LINKED ISSUE #157: Windows application without console has not stdin、stdout、stderr
hi， uvwasi requires three file descriptors to startup -- 0, 1, and 2（stdin、stdout、stderr）. Windows application without console has not stdin、stdout、stderr, I can open three files instead of them, but I don’t want to do this, is there any other solution? Thanks

### PR DESCRIPTION
**Pull request by phated**
When uvwasi is used on Windows, it might be used in non-"ConsoleMode" and the `uv_guess_handle` function can return two different values for `FILE_TYPE_CHAR` because of that difference, see https://github.com/libuv/libuv/blob/v1.x/src/win/handle.c#L42-L47

This function can also return `UV_FILE` in the case of `FILE_TYPE_DISK` but I assume that a disk file will properly stat and there isn't a good way to figure out which one is being returned.

This fixes #157 (took me about a month of debugging)

### CODE REVIEW

**Review by cjihrig — COMMENTED** [review]
Thanks for the patch. This LGTM with one small change. I think you'll also have to rebase this after https://github.com/nodejs/uvwasi/pull/180 gets approved and merged to take care of the CI.

**Comment by phated** [comment]
Thanks for the suggestion! I've merged it and I'll rebase once #180 lands.

**Comment by cjihrig** [comment]
It should be OK to rebase this now.

**Comment by phated** [comment]
Thanks @cjihrig! This has been rebased now. Cheers 🍻

**Review by cjihrig — APPROVED** [review]
LGTM. Thanks for the patch. Barring any surprises, this should be able to go out in the next Node 19 patch release.

**Comment by phated** [comment]
Excellent! We're actually using this through `wasm-micro-runtime` that uses uvwasi as an experimental WASI backend. And I'm writing bindings for OCaml, which is where I encountered this bug.

**Comment by cjihrig** [comment]
Oh that's cool. In that case, you won't have to wait for a Node release (which would probably be next year). I'll get a uvwasi release out today.

**Comment by cjihrig** [comment]
This is published in v0.0.14.

**Pull request merged**
